### PR TITLE
Add light-weight MLP

### PR DESCRIPTION
**Pull request by weihua916**

### CODE REVIEW

**Review by yiweny — APPROVED** [review]

File: torch_frame/nn/models/mlp.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        num_layers: int,
+        col_stats: dict[str, dict[StatType, Any]],
+        col_names_dict: dict[torch_frame.stype, list[str]],
+        stype_encoder_dict: dict[torch_frame.stype, StypeEncoder]
+        | None = None,
+        normalization: str | None = "layernorm",
+        dropout_prob: float = 0.2,
+    ) -> None:
+        super().__init__()
+        if num_layers <= 0:
```
[the remark this answers is not in the document]
Comment (weihua916) [reply torch_frame/nn/models/mlp.py:69]: Yeah we can delete this.

File: torch_frame/nn/models/mlp.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+        self.encoder = StypeWiseFeatureEncoder(
+            out_channels=channels,
+            col_stats=col_stats,
+            col_names_dict=col_names_dict,
+            stype_encoder_dict=stype_encoder_dict,
+        )
+
+        self.mlp = Sequential()
+        norm_cls = LayerNorm if normalization == "layernorm" else BatchNorm1d
```
Comment (rusty1s) [inline torch_frame/nn/models/mlp.py:84]: You don't respect no normalization here.

File: torch_frame/nn/models/mlp.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+            self.mlp.append(Dropout(p=dropout_prob))
+        self.mlp.append(Linear(channels, out_channels))
+
+        self.reset_parameters()
+
+    def reset_parameters(self) -> None:
+        self.encoder.reset_parameters()
+        for param in self.mlp:
+            if (isinstance(param, Linear) or isinstance(param, BatchNorm1d)
+                    or isinstance(param, LayerNorm)):
```
Comment (rusty1s) [inline torch_frame/nn/models/mlp.py:98]: if hasattr(self, 'reset_parameters'):

File: torch_frame/nn/models/mlp.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+class MLP(Module):
+    r"""The light-weight MLP model that mean-pools column embeddings and
+    applies MLP over it.
+
+    Args:
+        channels (int): The number of channels in the backbone layers.
+        out_channels (int): The number of output channels in the decoder.
+        num_layers (int): The number of layers in the backbone.
+        col_stats(dict[str,Dict[:class:`torch_frame.data.stats.StatType`,Any]]):
```
Comment (rusty1s) [inline torch_frame/nn/models/mlp.py:36]: missing white space.

**Pull request merged**